src/WebJobs.Script.WebHost/Metrics/HostMetricsProvider.cs

```diff
@@ -1,30 +1,48 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
-#nullable enable
 
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics.Metrics;
 using System.Linq;
 using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Script.Metrics;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
-namespace Microsoft.Azure.WebJobs.Script.Metrics
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Metrics
 {
     public class HostMetricsProvider : IHostMetricsProvider, IDisposable
     {
         private readonly MeterListener _meterListener;
         private readonly IServiceProvider _serviceProvider;
+        private readonly IOptionsMonitor<StandbyOptions> _standbyOptions;
+        private readonly ILogger<HostMetricsProvider> _logger;
+        private readonly object _lock = new object();
+
         private ConcurrentDictionary<string, long> _metricsCache = new();
```

Comment:
Does `_metricCache` needs to be  concurrent dictionary? Looks like all operations on that is done within a `lock` block.

Reply:
Furthermore, there is code below that is recreating it as a regular dictionary so yes this should be fixed

Reply:
Switching back to ConcurrentDictionary without a lock; Friday evening coding did me bad 😅